controller/pkg/claimsheader/bytes.go

```diff
@@ -0,0 +1,29 @@
+package claimsheader
+
+import (
+	"encoding/binary"
+
+	"go.aporeto.io/trireme-lib/controller/constants"
+)
+
+// HeaderBytes is the claimsheader in bytes
+type HeaderBytes []byte
+
+// ToClaimsHeader parses the bytes and returns the ClaimsHeader
+// WARNING: Caller has to make sure that headerbytes is NOT nil
+func (c HeaderBytes) ToClaimsHeader() *ClaimsHeader {
+
+	return &ClaimsHeader{
+		compressionType:  constants.CompressionTypeMask(c.extractHeaderAttribute(constants.CompressionTypeBitMask.ToUint8())),
+		encrypt:          uint8ToBool(c.extractHeaderAttribute(EncryptionEnabledMask)),
+		handshakeVersion: c.extractHeaderAttribute(HandshakeVersion),
+	}
+}
+
+// extractHeaderAttribute returns the attribute from bytes
+func (c HeaderBytes) extractHeaderAttribute(mask uint8) uint8 {
```

Comment:
mask has to 32 bits and data needs to be extracted as uint32.

Reply:
Fixed